### PR TITLE
Nullable System.ComponentModel, System.Diagnostics.CodeAnalysis, System.Diagnostics.Contracts

### PR DESCRIPTION
**Pull request by Anipik**

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/Diagnostics/Contracts/ContractFailedEventArgs.cs [hunk 1]
```diff
@@ -13,15 +14,15 @@ namespace System.Diagnostics.Contracts
...
+        private string? _condition;
+        private Exception? _originalException;
         private bool _handled;
         private bool _unwind;
 
-        internal Exception thrownDuringHandler;
+        internal Exception? thrownDuringHandler;
 
-        public ContractFailedEventArgs(ContractFailureKind failureKind, string message, string condition, Exception originalException)
+        public ContractFailedEventArgs(ContractFailureKind failureKind, string? message, string? condition, Exception originalException)
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Diagnostics/Contracts/ContractFailedEventArgs.cs]: Shouldn't originalException be `Exception?`?

File: src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs [hunk 2]
```diff
@@ -36,7 +37,7 @@ public DefaultValueAttribute(Type type, string value)
             // We would fail to load an otherwise normal class.
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs]: I believe null is a valid value for the `string value` parameter.

Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs]: It might even be for the `type` parameter.

Comment (Anipik) [reply src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs]: We have a catch statement to catch any type of exceptions but internally ```convertFromInvariantString``` throws when string or type is null

File: src/System.Private.CoreLib/shared/System/Diagnostics/Contracts/ContractFailedEventArgs.cs (hunk 1 above)
Comment (safern) [reply src/System.Private.CoreLib/shared/System/Diagnostics/Contracts/ContractFailedEventArgs.cs]: Yep, we even `Debug.Assert` for it to being `null`

File: src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs]: Yes, but that's an implementation detail.

**Pull request merged**